### PR TITLE
Use deterministic time progress in test

### PR DESCRIPTION
**Pull request by konradkonrad**
This is a proposal to tackle the flakyness ( #389 ) of
`raiden.tests.unit.test_transport.py::test_throttle_policy_ping`.

Instead of tracking actual sleep times this now just tests for the correct
calls to `gevent.sleep` in the throttle policy implementation.

### CODE REVIEW

File: raiden/tests/unit/test_transport.py [hunk 1]
```diff
@@ -74,14 +93,16 @@ def test_throttle_policy_ping(monkeypatch, raiden_network):
...
     assert len(messages) == 4  # two Pings and the corresponding Acks
 
     # Now check the fill_rate for the remaining packets
     for i in range(2, 10):
         events[i].wait()
-        check.elapsed(token_refill, tolerance)
+        test_time.progress(token_refill)
+
+    assert all(t == 0.0 for t in sleeps.sleeptimes[:4]), sleeps.sleeptimes
+    assert all(t == 0.5 for t in sleeps.sleeptimes[4:]), sleeps.sleeptimes
```
Comment (hackaugusto) [inline raiden/tests/unit/test_transport.py]: Why 0.5? is it the `token_refill` value?

Comment (konradkonrad) [inline raiden/tests/unit/test_transport.py]: yes, I changed it.

File: raiden/tests/unit/test_transport.py [hunk 2]
```diff
@@ -74,14 +93,16 @@ def test_throttle_policy_ping(monkeypatch, raiden_network):
...
 
     assert len(messages) == 4  # two Pings and the corresponding Acks
 
     # Now check the fill_rate for the remaining packets
     for i in range(2, 10):
         events[i].wait()
-        check.elapsed(token_refill, tolerance)
+        test_time.progress(token_refill)
+
+    assert all(t == 0.0 for t in sleeps.sleeptimes[:4]), sleeps.sleeptimes
```
Comment (hackaugusto) [inline raiden/tests/unit/test_transport.py]: Why the slice uses `4`? Does `sleeps.sleeptimes` include an additional sleep per *sent* Ping? Perhaps for the Ack?

Comment (konradkonrad) [inline raiden/tests/unit/test_transport.py]: commented --- it is app0: Ping Ping (unthrottled) and app1: Ack Ack (unthrottled)

**Pull request merged**